modules/apps/redirect/redirect-web/src/main/java/com/liferay/redirect/web/internal/portlet/action/EditRedirectPatternsMVCActionCommand.java

```diff
@@ -110,10 +114,19 @@ private List<RedirectPatternEntry> _getRedirectPatternEntries(
 			if ((patternString != null) && (destinationURL != null) &&
 				(userAgent != null)) {
 
-				redirectPatternEntries.add(
-					new RedirectPatternEntry(
-						Pattern.compile(patternString), destinationURL,
-						userAgent));
+				try {
+					redirectPatternEntries.add(
+						new RedirectPatternEntry(
+							Pattern.compile(patternString), destinationURL,
+							userAgent));
+				}
+				catch (PatternSyntaxException patternSyntaxException) {
+					_log.error(patternSyntaxException);
```

Comment:
I don't know what's the general behaviour.... should we log a validation problem as an error?